### DISCUSSION TITLE
Snapshots: Support snapshots of dashboards where some panels are not visible (scroll down to see)

### DISCUSSION THREAD

**Discussion by lunaticusgreen** [grafana/grafana#38198] [Dashboards, Panels & Visualizations]
**What happened**:
I created a dashboard snapshot and data were only available for those nested dashboards that were visible and got data by the moment of snapshot creation. as for this snapshot https://snapshot.raintank.io/dashboard/snapshot/6zQO23v14Q3n2OAXuAsxOLd0LCtrHbnj?orgId=2 ( Check **Traffic In/Out** dashboard )

**What you expected to happen**:
Snapshot has data for all dashboards on the page, as for this snapshot https://snapshot.raintank.io/dashboard/snapshot/3BUOcaS3aPXgUOZb33DGfeBgV8JPD7oc?orgId=2 ( Check **Traffic In/Out** dashboard )

**How to reproduce it (as minimally and precisely as possible)**:

1. Go to https://play.grafana.org/d/000000012/grafana-play-home?orgId=1
2. Click to "Share" icon, go to the snapshot tab and create a snapshot with "Publish to snapshot.raintank.io"

**Anything else we need to know?**:

As far as I can understand, the reason for this behavior is the lazy loading that was added in Grafana v6.2

**Environment**:
- Grafana version: latest
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser: MacOS & Firefox developer edition 85.0b4
- Grafana plugins:
- Others:

**Comment by torkelo** [comment]
Sorry added wrong label

**Comment by rnovikovP** [comment]
Hi @torkelo any ideas/recommendations on where to look and how to approach with a fix on this. We would like to contribute to this fix

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by MrPowerGamerBR** [comment]
Can confirm this issue, this breaks snapshot's functionalities because you need to forcefully increase your browser window to include all panels in your window, and *then* you can create a snapshot.

If you don't do this, only the visible panels will be stored in the snapshot. Scrolling down to load panels doesn't work because it seems like Grafana also unloads panels out of view.

**Reply by MrPowerGamerBR** [reply]
After looking at the code... @rnovikov if you are still interested in fixing this issue, what could be done is changing the `Share Snapshot` code to force all panels to be loaded (https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/components/ShareModal/ShareSnapshot.tsx#L81), the code that causes the panels to be unloaded is here (https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/dashgrid/DashboardGrid.tsx#L126)

So maybe a `shouldBeKeptLoaded` property should be added to the `DashboardGrid`, and then the `ShareSnapshot` code could toggle that property when a shapshot is being taken.

But IMO the proper solution would be for the snapshot code to query all panels manually, instead of relying in something that feels like a hack.

**Reply by ademidoff** [reply]
Whatever solution the Grafana team lands to fix it will be good. Right now it's not clear why part of the panels do not get any data at all :(

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82952. Feel free to continue the discussion around this there. Thank you!
